media-proxy/src/rdma_session.c

```diff
@@ -345,6 +345,11 @@ static void *rx_rdma_frame_thread(void *arg)
     libfabric_ctx *rdma_ctx = s_ctx->rdma_ctx;
     ep_ctx_t *cp_ctx = s_ctx->ep_ctx;
 
+    while (!atomic_load(&s_ctx->shm_ready)) {
```

Comment:
If the user or the system decides to shutdown the app, how do you exit this loop gracefully?

Reply:
currently [when kill signal is received exit(0) is called ](https://github.com/OpenVisualCloud/Media-Communications-Mesh/blob/main/media-proxy/src/api_server_tcp.cc#L270) which terminates all threads.

It is not the most elegant way, I agree, but I don't see any possible critical outcomes.

Reply:
Not handling graceful shutdown brings the following potential issues:
* Memory leaks
* System resources are not released (including networking sockets).
* 3rd party software components' resources are not released.
For now, I'm okay with using the existing approach of relying on `exit(1)`. But this must be reworked in the future to support graceful shutdown.

Please leave a comment here, something like `TODO: Implement graceful shutdown.`

Reply:
Ok. I agree that we can leave this comment. This will be a good improvement, but I'm not sure if any memory leak or unreleased resources are possible. I think OS performs cleanup after a process is terminated and it release all resources that the process acquired.

Reply:
Comment is added.

Reply:
You know what. I removed the comment and added `&& !s_ctx->stop` to the condition. This is our graceful shutdown. We can stop this thread by setting the `stop` variable. It does not acquire any resources so exiting the loops is enough.

Another thing is if  the `stop` variable is well synchronized, in my view it is not, but maybe it is something to fix in different PR.